src/test/java/org/folio/circulation/resources/renewal/OverrideRenewalStrategyTest.java

```diff
@@ -0,0 +1,361 @@
+
+    final RenewalContext renewalContext = RenewalContext.create(loan, overrideRequest, "no-user")
+      .withRequestQueue(new RequestQueue(emptyList()));
+
+    return renew(renewalContext);
+  }
+
+  private Result<Loan> renew(RenewalContext context) {
+    return new OverrideRenewalStrategy().renew(context, null)
+      .getNow(Result.failed(new ServerErrorFailure("Failure")))
+      .map(RenewalContext::getLoan);
+  }
+
+  private Result<Loan> renewWithRecall(Loan loan, DateTime overrideDueDate) {
+    final JsonObject overrideRequest = createOverrideRequest(overrideDueDate);
+
+    final RenewalContext renewalContext = RenewalContext.create(loan, overrideRequest, "no-user")
+      .withRequestQueue(new RequestQueue(singleton(createRecallRequest())));
+
+    return renew(renewalContext);
+  }
+
+  private Request createRecallRequest() {
+    return Request.from(new JsonObject().put("requestType", "Recall"));
+  }
+
+  private JsonObject createOverrideRequest(DateTime dueDate) {
+    final JsonObject json = new JsonObject();
+
+    write(json, "comment", "A comment");
+    write(json, "dueDate", dueDate);
+
+    return json;
+  }
+
+  private Item createDeclaredLostItem() {
+    final JsonObject json = new JsonObject();
+
+    writeByPath(json, "Declared lost", "status", "name");
+
+    return Item.from(json);
+  }
+
+  private Loan createLoanWithDueDateAfterProposed() {
+    return createLoanWithDefaultPolicy()
+      .changeDueDate(now(UTC).plusDays(8));
+  }
+
+  private Loan createLoanWithDefaultPolicy() {
+    final Item checkedOutItem = Item.from(new JsonObject()
+      .put("status", new JsonObject().put("name", "Checked out")));
+
+    return Loan.from(new JsonObject())
+      .withItem(checkedOutItem)
+      .withLoanPolicy(LoanPolicy.from(rollingPolicy().create()));
+  }
+
+  private void assertDueDate(DateTime expected, Result<Loan> actual) {
+    assertTrue(actual.succeeded());
+    assertNotNull(actual.value());
+    assertNotNull(actual.value().getDueDate());
+
+    assertEquals(expected.getMillis(), actual.value().getDueDate().getMillis());
+  }
+
+  private void assertDueDateWithinOneSecondAfter(DateTime after, Result<Loan> actual) {
+    assertTrue(actual.succeeded());
+    assertNotNull(actual.value());
+    assertNotNull(actual.value().getDueDate());
+
+    assertThat(actual.value().getDueDate().toString(), withinSecondsAfter(seconds(1), after));
+  }
+
+  private void assertValidationError(Result<Loan> renewResult, String expectedMessage) {
+    assertTrue(renewResult.failed());
+    assertTrue(renewResult.cause() instanceof ValidationErrorFailure);
+
+    final ValidationErrorFailure failure = (ValidationErrorFailure) renewResult.cause();
+    if (!failure.hasErrorWithReason(expectedMessage)) {
+      System.out.println("Expected: " + expectedMessage);
+      System.out.println("Actual: " + failure.getErrors().iterator().next().getMessage());
+    }
+    assertTrue(failure.toString().contains(expectedMessage));
```

Comment:
This seems very similar to code in another class could these be combined into a matcher (or set of matchers)?